hackathon_site/hardware/serializers.py

```diff
@@ -46,9 +54,128 @@ def get_unique_hardware_count(obj: Category):
         return Hardware.objects.filter(categories__id=obj.id).count()
 
 
-class OrderSerializer(serializers.ModelSerializer):
+class OrderListSerializer(serializers.ModelSerializer):
     hardware_set = HardwareSerializer(many=True, read_only=True)
 
     class Meta:
         model = Order
         fields = ("id", "hardware_set", "team", "status", "created_at", "updated_at")
+
+
+class OrderCreateSerializer(serializers.Serializer):
+    class OrderCreateHardwareSerializer(serializers.Serializer):
+        id = serializers.PrimaryKeyRelatedField(
+            queryset=Hardware.objects.all(), many=False, required=True
+        )
+        quantity = serializers.IntegerField(required=True)
+
+    hardware = OrderCreateHardwareSerializer(many=True, required=True)
+
+    @staticmethod
+    def merge_requests(hardware_requests):
+        return functools.reduce(
+            lambda x, y: x + y,
+            [Counter({e["id"]: e["quantity"]}) for e in hardware_requests],
+        )
+
+    # check that the requests are within per-team constraints
+    def validate(self, data):
+        requested_hardware = self.merge_requests(hardware_requests=data["hardware"])
```

Comment:
This is just a counter (dict) mapping id -> quantity, right? If so, the rest of your logic assumes the keys are objects not ints